core/include/geometry/volume.hpp

```diff
@@ -90,23 +100,26 @@ class volume {
     }
 
     /** @return range of surfaces by surface type - const access */
+
     template <typename object_type>
-    inline constexpr const auto &range() const {
+    DETRAY_HOST_DEVICE inline constexpr const auto &range() const {
         constexpr auto index = object_registry::template get<object_type>();
         return std::get<index>(_ranges);
     }
 
     /** @return range of surfaces- const access */
     template <
         typename object_registry::id range_id = object_registry::id::e_surface>
-    inline constexpr const auto &range() const {
+    DETRAY_HOST_DEVICE inline constexpr const auto &range() const {
         return std::get<range_id>(_ranges);
     }
 
-    private:
+    /** @return _ranges */
+    inline const auto &ranges() const { return _ranges; }
```

Comment:
This is only needed in the geometry unittest? I don't think we should keep it then

Reply:
Well OK :)